sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4773,10 +4777,23 @@ enqueueMemFillHelper(pi_command_type CommandType, pi_queue Queue, void *Ptr,
   // Get a new command list to be used on this call
   ze_command_list_handle_t ZeCommandList = nullptr;
   ze_fence_handle_t ZeFence = nullptr;
+
   // Performance analysis on a simple SYCL data "fill" test shows copy engine
   // is faster than compute engine for such operations.
+  bool PreferCopyEngine = Queue->Device->hasCopyEngine();
+  if (PreferCopyEngine) {
+    // Make sure that pattern size matches the capability of the queue.
+    if (PatternSize >
+        Queue->Device->ZeCopyQueueGroupProperties.maxMemoryFillPatternSize) {
+      PreferCopyEngine = false;
+      PI_ASSERT(PatternSize <= Queue->Device->ZeComputeQueueGroupProperties
```

Comment:
I think this should be 'less than' based on doc.

Reply:
Level-Zero header file says:
``` C++
size_t maxMemoryFillPatternSize;                ///< [out] maximum `pattern_size` supported by command queue group.
```

Reply:
Hmm..the SPEC doc says
The pattern size must be a power-of-two and less than ze_command_queue_group_properties_t.maxMemoryFillPatternSize.
I am looking at the SPEC doc from here:
https://spec.oneapi.com/level-zero/latest/core/api.html#ze__api_8h_1a689ec654289f637226b9d6e84a42473b
Thanks

Reply:
Right, there is a disconnect between the header and the spec.
In my case, implementation reports 16 for maximum on copy-queue, and given that it must also be a power of 2, it does not make sense to be "less than 16".

I am clarifying with the spec owners.

Reply:
We want to "max" be inclusive value, and will adjust the spec. So code here is good as is.